policy-controller/k8s/index/src/inbound/index.rs

```diff
@@ -1760,6 +1868,65 @@ impl PolicyIndex {
         authzs
     }
 
+    fn client_ratelimit(&self, server_name: &str) -> Option<RateLimit> {
+        for (name, spec) in self.ratelimit_policies.iter() {
```

Comment:
This seems like it might be non-deterministic if there are multiple policies targetting a server. Since we're iterating over a HashMap, we have no guarantees about the iteration order.

Are there guarantees elsewhere that there is only one policy for a given server? Or how do we deal with ambiguity?

Reply:
Ah, I don't think that's sufficient: we can't rely on admission controllers for any inter-resource validation constraints (like uniqueness). When applying multiple resources, for instances, the validator's view of the cluster may lag the reality. Or the validator may not be invoked for certain resources--resources may be applied when the validator was down or the config was not valid. Admission controllers are only useful to validate that a resource is _internally_ consistent--i.e. to apply schema validation.

This is why we use statuses (e.g. for HTTPRoutes) to control 'acceptance'--we must accept that resources may be created in the cluster, and we must have an eventually consistent view of those that are actually admitted by the controller.

Reply:
I expect that the best path forward is:

1. In this PR, we should determine/document precedence for these policies (probably using the same rules for HTTPRoute attachment). If the controller honors these precedence rules, then we can at least ensure that there is predictable/consistent behavior.
2. We should follow up by adding statuses to the policy resources that communicate the acceptance status of the resource.

Reply:
Ok in 21e74d633 I've changed this particular function to only use the first RL by creation time and name. This means we keep the model to be one-to-one between servers and RLs, and we're not combining rules in any way.